### ISSUE TITLE
High CPU and Ram on Synology NAS

### ISSUE THREAD

**Issue by Shasoosh** [Celedhrim/docker-kodi-server#23]
Hey

I've just run the container and without it doing anything special, I'm seeing constant 15% CPU and 1.2GB of RAM usage. 

Is that normal behavior ?

![image](https://user-images.githubusercontent.com/2679978/98449927-b4361980-2140-11eb-8c17-c8bdb913f6c4.png)

**Comment by Celedhrim** [comment]
Mine take 500mb memory and ~3% cpu when idle so i don't know :/

**Comment by Shasoosh** [comment]
Any idea what might affect it?
I'm using a DS920+ Synology NAS with +-200 Movies.

btw, what is the CPU used for when idle?

**Comment by Celedhrim** [comment]
Look at the README , i've just update it.

You could try disable all menu to not display anything dynamically ( like last movies , random movies etc , also , hide library update)
Anything that move on the GUI cause fake display using CPU

**Comment by DAVIZINH0** [comment]
hello,
this is the one container with kodi matrix, so CONGRATULATION! AND thanks!!!!

I mount it in a synology and have a heavy cpu usage 15 iddle. 50 max (synology 1819+), ram is razonable (500mb aprox).
But i have a lot of reboots of the container and i dont know why.

maybe it will be interesting make a list of "guisettings.xml" with a recomend settings to maximice de performance.

**Comment by Celedhrim** [comment]
Thx :)
xpra is not very good for performance , but at least it's works without doing crazing things to kodi :)

Mine is up for 4 days , use 2% cpu on idle ( eat on start and database clean because display do something) and ~600 MB of memory.

did you have log while is failed ? oom ? segfault ?

**Comment by DAVIZINH0** [comment]
totally agree this is best than nothing!!! thanks a lot!!! really thanks!!!

this is my log
[X.log](https://github.com/Celedhrim/docker-kodi-server/files/6043658/X.log)

and this is a capture of the reboots
<img width="1307" alt="reboots" src="https://user-images.githubusercontent.com/32111460/109168222-e23a7a80-777e-11eb-8816-7b076baf128f.png">

**Comment by Celedhrim** [comment]
Does your image is the ubuntu 18.04 or my fail test on 20.04  ?

**Comment by DAVIZINH0** [comment]
> 
> 
> Does your image is the ubuntu 18.04 or my fail test on 20.04 ?

latest tag, download this morning. how can i see to give you the exact information?

**Comment by Celedhrim** [comment]
For pulseaudio I have this : 

```
2021-02-20 20:54:13,397 pulseaudio server started with pid 98
W: [pulseaudio] main.c: This program is not intended to be run as root (unless --system is specified).
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
E: [pulseaudio] core-util.c: Failed to connect to system bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
W: [pulseaudio] authkey.c: Failed to open cookie file '/root/.config/pulse/cookie': No such file or directory
W: [pulseaudio] authkey.c: Failed to load authentication key '/root/.config/pulse/cookie': No such file or directory
W: [pulseaudio] authkey.c: Failed to open cookie file '/root/.pulse-cookie': No such file or directory
W: [pulseaudio] authkey.c: Failed to load authentication key '/root/.pulse-cookie': No such file or directory
W: [pulseaudio] server-lookup.c: Unable to contact D-Bus: org.freedesktop.DBus.Error.Spawn.ExecFailed: /usr/bin/dbus-launch terminated abnormally without any error message
W: [pulseaudio] main.c: Unable to contact D-Bus: org.freedesktop.DBus.Error.Spawn.ExecFailed: /usr/bin/dbus-launch terminated abnormally without any error message
```
at startup and nothing more

Not like you.

to get os version in container :

```
celed@server:~$ docker ps | grep kodi
f230caa994f1   celedhrim/kodi-server:matrix               "/bin/kodi-run-wrapp…"   4 days ago     Up 4 days             8089/tcp, 9090/tcp, 9777/udp   kodi_kodi.1.1pafvd9koyto9wu46s2g1dpoe
celed@server:~$ docker exec -ti f230caa994f1 /bin/bash
root@f230caa994f1:/# cat /etc/issue
Ubuntu 18.04.5 LTS \n \l

root@f230caa994f1:/# exit
```

**Comment by DAVIZINH0** [comment]
same as you
`root@kodi-matrix:/# cat /etc/issue

Ubuntu 18.04.5 LTS \n \l

root@kodi-matrix:/# `

**Comment by Celedhrim** [comment]
Maybe something with synology docker implemntation , I don't know :(

**Comment by DAVIZINH0** [comment]
maybe, dont worry! i try the next days and monitor the reboots. thanks for your work!
Now the cpu usage is stable in 10%. is acceptable for me

**Comment by Shasoosh** [comment]
I ended up moving to another built: https://github.com/schwensk/kodi-headless
With it I'm getting less than 1% of CPU and a very low mem usage.
```
                     CPU                  RAM
kodi-headless       0.96%               29.51MiB / 19.39GiB   
```

**Comment by Celedhrim** [comment]
> 
> 
> I ended up moving to another built: https://github.com/schwensk/kodi-headless
> With it I'm getting less than 1% of CPU and a very low mem usage.
> 
> ```
>                      CPU                  RAM
> kodi-headless       0.96%               29.51MiB / 19.39GiB   
> ```

Yep , but it's another "headless" patch build. So only Leia until someone make a matrix headless patch :)

For me it's not future proof . My way to do this is really not ideal, I know that , but I can use stock  kodi from team kodi ppa .

**Comment by DAVIZINH0** [comment]
> 
> 
> I ended up moving to another built: https://github.com/schwensk/kodi-headless
> With it I'm getting less than 1% of CPU and a very low mem usage.
> 
> ```
>                      CPU                  RAM
> kodi-headless       0.96%               29.51MiB / 19.39GiB   
> ```

but this is not matrix, or im wrong?

**Comment by paulccs** [comment]
I'm having trouble connecting to MySQL from kodi in the container.  In the kodi.log it has a line like the following for every database version:
ERROR <general>: Unable to open database: MyVideos119 [2026](SSL connection error: unknown error number)

After running docker exec /bin/bash in the container, I did apt install mysql-client from within the container and was able to connect successfully using the ip 10.8.0.xxx.

I tried running update-ca-certificates from within the container and that didn't help with kodi.

The .kodi (portable_data) directory was copied from an Ubuntu version 20.04 system and I know this build is 18.04.  I didn't think that would be an issue, but is it?  Any ideas anyone?

(I've upgraded all my x86_64 18.04 systems to 20.04.)  Thanks!

**Comment by DAVIZINH0** [comment]
hi again, i see that a import creates all the thumbnails in the docker container.
Maybe disable this option, reduce the cpu usage when scan?

maybe its a stupid idea

**Comment by alek-05** [comment]
Hello,
I'm more or less in the same situation as Shasoosh: Synology DS1515+, around 200 movies and 70/80 TV series. The container use 12/13% cpu all time and only 300M ram. As you suggested I disabled all menu and set Settings as default. Not sure if I should do something else. Thank you.
![image](https://user-images.githubusercontent.com/44298932/110247765-22171400-7f6e-11eb-9dcd-22bd7702b028.png)

**Comment by desmoloch** [comment]
thanks for your work. I updated to matrix on my synology ds918+ and all is woring fine!
My cpu usage is about 4 % and using ~350 MB ram.
First i had about ~40 % CPU Usage, here is what i did:
- disable event logging
- setting gui resolution to 640x480 (my default resolution is 2560x1444)
i would suggest that my solution was the resolution :)

**Comment by alek-05** [comment]
I'm not able to save these settings from the webgui! I change it but when I save it doesn't do anything. And refreshing the page I see the default settings. Any suggestion on how to save it? Maybe from the xml files?
Thank you.

**Comment by desmoloch** [comment]
i changed it on my local (windows) kodi and transferred the configs.
Here is my config from guisettings.xml:
`<setting id="videoscreen.screenmode">0064000480056.00000pstd</setting> `
Edit it and it should work :)

**Comment by alek-05** [comment]
wow, it worked! :D many thanks! CPU is now around 5%, without disabling the event logging!
